The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/24630638138/job/72017313630) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/72017313630) [(plain)](https://github.com/rust-lang/rust/commit/7691a07b251e3081d16671c4ed155847cf329ff5/checks/72017313630/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
fmt: checked 6808 files
tidy check
tidy [rustdoc_json (src)]: `rustdoc-json-types` modified, checking format version
tidy: Skipping binary file check, read-only filesystem
tidy [style (tests)]: /checkout/tests/run-make/macho-link-section/rmake.rs:19: line longer than 100 chars
tidy [style (tests)]: /checkout/tests/run-make/macho-link-section/foo.rs:6: line longer than 100 chars
tidy [style (tests)]: FAIL
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (26.0.1)
linting python files
---
info: ES-Check: there were no ES version matching errors!  🎉
typechecking javascript files
tidy: The following check failed: style (tests)
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy --root-path=/checkout --cargo-path=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo --output-dir=/checkout/obj/build --concurrency=4 --npm-path=/node/bin/yarn --ci=true --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1618:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1417:29

--- BACKTRACE vvv
   0: <bootstrap::utils::exec::DeferredCommand>::finish_process
             at /checkout/src/bootstrap/src/utils/exec.rs:939:17
   1: <bootstrap::utils::exec::DeferredCommand>::wait_for_output::<&bootstrap::utils::exec::ExecutionContext>
             at /checkout/src/bootstrap/src/utils/exec.rs:831:21
   2: <bootstrap::utils::exec::ExecutionContext>::run
             at /checkout/src/bootstrap/src/utils/exec.rs:741:45
   3: <bootstrap::utils::exec::BootstrapCommand>::run::<&bootstrap::core::builder::Builder>
             at /checkout/src/bootstrap/src/utils/exec.rs:339:27
   4: <bootstrap::core::build_steps::test::Tidy as bootstrap::core::builder::Step>::run
             at /checkout/src/bootstrap/src/core/build_steps/test.rs:1417:29
   5: <bootstrap::core::builder::Builder>::ensure::<bootstrap::core::build_steps::test::Tidy>
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1579:36
   6: <bootstrap::core::build_steps::test::Tidy as bootstrap::core::builder::Step>::make_run
             at /checkout/src/bootstrap/src/core/build_steps/test.rs:1339:21
   7: <bootstrap::core::builder::StepDescription>::maybe_run
             at /checkout/src/bootstrap/src/core/builder/mod.rs:476:13
   8: bootstrap::core::builder::cli_paths::match_paths_to_steps_and_run
             at /checkout/src/bootstrap/src/core/builder/cli_paths.rs:232:18
   9: <bootstrap::core::builder::Builder>::run_step_descriptions
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1122:9
  10: <bootstrap::core::builder::Builder>::execute_cli
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1101:14
  11: <bootstrap::Build>::build
             at /checkout/src/bootstrap/src/lib.rs:799:25
  12: bootstrap::main
             at /checkout/src/bootstrap/src/bin/main.rs:130:11
  13: <fn() as core::ops::function::FnOnce<()>>::call_once
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/core/src/ops/function.rs:250:5
  14: std::sys::backtrace::__rust_begin_short_backtrace::<fn(), ()>
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/sys/backtrace.rs:166:18
  15: std::rt::lang_start::<()>::{closure#0}
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/rt.rs:206:18
  16: <&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe as core::ops::function::FnOnce<()>>::call_once
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/core/src/ops/function.rs:287:21
  17: std::panicking::catch_unwind::do_call::<&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe, i32>
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/panicking.rs:581:40
  18: std::panicking::catch_unwind::<i32, &dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe>
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/panicking.rs:544:19
  19: std::panic::catch_unwind::<&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe, i32>
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/panic.rs:359:14
  20: std::rt::lang_start_internal::{closure#0}
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/rt.rs:175:24
  21: std::panicking::catch_unwind::do_call::<std::rt::lang_start_internal::{closure#0}, isize>
             at /rustc/ef0fb8a2563200e322fa4419f09f65a63742038c/library/std/src/panicking.rs:581:40
---
  28: __libc_start_main
  29: _start


Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:02:54
  local time: Sun Apr 19 13:56:21 UTC 2026
  network time: Sun, 19 Apr 2026 13:56:22 GMT
##[error]Process completed with exit code 1.
##[group]Run echo "disk usage:"
```

</details>

--- a/tests/run-make/macho-link-section/foo.rs
+++ b/tests/run-make/macho-link-section/foo.rs
@@ -1,0 +1,14 @@
+#[unsafe(no_mangle)]
+#[unsafe(link_section = "__TEXT,custom_code,regular,pure_instructions")]
+static CODE: [u8; 10] = *b"0123456789";
+
+#[unsafe(no_mangle)]
+#[unsafe(link_section = "__DATA,all_attributes,regular,pure_instructions\
+                        +no_toc+strip_static_syms+no_dead_strip+live_support\
+                        +self_modifying_code+debug")]
+static ALL_THE_ATTRIBUTES: u32 = 42;
+
+#[unsafe(no_mangle)]
+#[unsafe(link_section = "__DATA,__mod_init_func,mod_init_funcs")]
+static CONSTRUCTOR: extern "C" fn() = constructor;
+extern "C" fn constructor() {}

--- a/tests/run-make/macho-link-section/rmake.rs
+++ b/tests/run-make/macho-link-section/rmake.rs
@@ -1,0 +1,47 @@
+//! Test that various Mach-O `#[link_section]` values are parsed and passed on correctly by codegen
+//! backends.
+//@ only-apple
+use run_make_support::{llvm_objdump, rustc};
+
+fn main() {
+    rustc().input("foo.rs").crate_type("lib").arg("--emit=obj").run();
+
+    let stdout =
+        llvm_objdump().arg("--macho").arg("--private-headers").input("foo.o").run().stdout_utf8();
+
+    let expected = [
+        ("__TEXT", "custom_code", "S_REGULAR", "PURE_INSTRUCTIONS"),
+        ("__DATA", "__mod_init_func", "S_MOD_INIT_FUNC_POINTERS", "(none)"),
+        (
+            "__DATA",
+            "all_attributes",
+            "S_REGULAR",
+            "PURE_INSTRUCTIONS NO_TOC STRIP_STATIC_SYMS \
+             NO_DEAD_STRIP LIVE_SUPPORT SELF_MODIFYING_CODE DEBUG",
+        ),
+    ];
+
+    for (segment, section, section_type, section_attributes) in expected {
+        let mut found = false;
+        // Skip header.
+        for section_info in stdout.split("Section").skip(1) {
+            if section_info.contains(&format!("segname {segment}"))
+                && section_info.contains(&format!("sectname {section}"))
+            {
+                assert!(
+                    section_info.contains(&format!("type {section_type}")),
+                    "should have type {section_type:?}"
+                );
+                assert!(
+                    section_info.contains(&format!("attributes {section_attributes}\n")),
+                    "should have attributes {section_attributes:?}"
+                );
+                found = true;
+            }
+        }
+
+        if !found {
+            panic!("could not find section {section} in binary");
+        }
+    }
+}

